src/core/SP_reader.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: GNU AGPLv3
+pragma solidity ^0.8.20;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Attestation as SPAttestation } from "../models/Attestation.sol"; // Ignoring Offchain Attestations
+import { Attestation as VeraxAttestation, IVerax } from "src/interfaces/IVerax.sol";
+import { SP } from "src/core/SP.sol";
+/*
+    Found Information Above on Github:
+    - Verax's Attestation struct 
+    - Interface I created to call a certain function -- getAttestation()
+*/
+
+contract Reader is OwnableUpgradeable {
+    /// @custom:storage-location erc7201:ethsign.SP-Reader
+    struct ReaderStorage {
+        SP signProtocol;
+        IVerax veraxRegistry;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("ethsign.SP-Reader")) - 1)) & ~bytes32(uint256(0xff))
```

Comment:
What's inside the `keccak256` is a variant of this: https://en.wikipedia.org/wiki/Reverse_domain_name_notation, so try to follow the rules, e.g. no `-` in the naming.